src/Appwrite/Specification/Format/Swagger2.php

```diff
@@ -17,15 +17,32 @@ public function getName(): string
     protected function getNestedModels(Model $model, array &$usedModels): void
     {
         foreach ($model->getRules() as $rule) {
-            if (
-                in_array($model->getType(), $usedModels)
-                && !in_array($rule['type'], ['string', 'integer', 'boolean', 'json', 'float', 'double'])
-            ) {
-                $usedModels[] = $rule['type'];
-                foreach ($this->models as $m) {
-                    if ($m->getType() === $rule['type']) {
-                        $this->getNestedModels($m, $usedModels);
-                        return;
+            if (!in_array($model->getType(), $usedModels)) {
```

Comment:
Don't we already have support for multiple types? why do we need this change?
Also, if required, we should probably have the same change for OpenAPI3

Reply:
Honestly, hard to tell.. I remember sitting with Torsten trying to merge my changes with his (nested models) together. This is the outcome.

I tried to remove these changes and use what is there in 0.16.x, but then specs were missing some hash types:
![CleanShot 2022-08-16 at 13 34 35](https://user-images.githubusercontent.com/19310830/184870056-aed89241-bae9-4925-a208-7cf687718e54.png)

The thing is, this PR creates a new edge-case when a model (definition) is not used directly in any endpoint, instead, is used by a model that is being used in an endpoint.
To be specific, there is "Get User" that returns "User" model. and on "User" there is "hashOptions" that is oneOf AlgoArgon, AlgoScrypt, AlgoBCrypt...

This change to Swagger2 adds support for oneOf/anyOf models that are not directly used in an endpoint, instead, they are mentioned in a different model.

Reply:
You are right we need the same change in Swagger3. Added. I copy&pasted with the only difference being support for `double` in Swagger3.

Reply:
OK, but we should probably replicate this to OPENAPI3

Reply:
My bad, that's what I meant. I just added it to OpenAPI3.